### PR TITLE
Handle SSL errors gracefully

### PR DESCRIPTION
**Pull request by jaceklyp**
Task/Issue URL: https://app.asana.com/0/72649045549333/1207295365843956/f
iOS PR: https://github.com/duckduckgo/iOS/pull/3271
macOS PR: TBD
What kind of version bump will this require?: Major

Tech Design URL:
https://app.asana.com/0/1203790657351911/1207778710311430/f
https://app.asana.com/0/481882893211075/1208111823478404/f

**Description**:
If users visit a page with certificate or other security issues, provide warning information and allow users to proceed (if they choose to).

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/SSLErrors/ErrorPageHTMLTemplate.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import ContentScopeScripts
+import WebKit
+import Common
+
+public struct ErrorPageHTMLTemplate {
```
Comment (SabrinaTardio) [inline Sources/SSLErrors/ErrorPageHTMLTemplate.swift:24]: We should call this SpecialErrorPageHTMLTemplate cause in macOS ErrorPageHTMLTemplate is a different struct

File: Sources/PrivacyDashboard/PrivacyDashboardController.swift [hunk 2]
```diff
@@ -244,7 +244,11 @@ extension PrivacyDashboardController: WKNavigationDelegate {
         privacyInfo?.$serverTrust
             .receive(on: DispatchQueue.global(qos: .userInitiated))
             .map { serverTrust in
-                ServerTrustViewModel(serverTrust: serverTrust)
+                if let serverTrust {
+                    // swiftlint:disable:next force_cast
+                    return ServerTrustViewModel(serverTrust: (serverTrust as! SecTrust))
+                }
+                return ServerTrustViewModel(serverTrust: nil)
```
Comment (SabrinaTardio) [inline Sources/PrivacyDashboard/PrivacyDashboardController.swift:251]: I am unsure why we are doing this?

**Review by SabrinaTardio — COMMENTED** [review]
Thanks for extracting common code!
I left some comment! if you address them i can work on the macOS PR (or you can have a shot if you have time)

File: Sources/PrivacyDashboard/PrivacyDashboardController.swift (hunk 2 above)
Comment (jaceklyp) [reply Sources/PrivacyDashboard/PrivacyDashboardController.swift:251]: so we can properly mock the secTrust in tests, I couldn't get away without warning or error hence this code, if you know a better way I'll be happy to simplify it!

File: Sources/SSLErrors/SpecialErrorPageUserScript.swift [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+typealias Key = String
+
+struct LocalizedInfo: Encodable, Equatable {
+
+    let title: String
+    let note: String
+
+}
+
+public final class SpecialErrorPageUserScript: NSObject, Subfeature {
```
[the remark this answers is not in the document]
Comment (jaceklyp) [reply Sources/SSLErrors/SpecialErrorPageUserScript.swift:41]: yeah, saw them, would be perfect!

**Review by SabrinaTardio — COMMENTED** [review]
LGTM! Can you fix the last test seems failing?

**Review by SabrinaTardio — APPROVED** [review]
LGTM!

**Pull request merged**